### DISCUSSION TITLE
How to deduplication @tailwindbase indication in a monorepo environment

### DISCUSSION THREAD

**Discussion by KIMSEUNGGYU** [tailwindlabs/tailwindcss#14156] [Help]
Currently, monorepo is being used in the tuborepo environment.

A brief description of the project structure is the apps/accounts environment and the packages/sds (ui) environment.

Both sds and accounts style using tailwind.

### Problem
Q1. 
Currently, sds has the following styles and accounts are in the same environment.
```css
<!-- apps/accounts/styles/global.css -->
@import '../components/sds/input/InputField/input.style.css';

@tailwind base;
@tailwind components;
@tailwind utilities;

<!-- other style -->
```

```css
<!-- packages/sds/styles/global.css -->
@tailwind base;
@tailwind components;
@tailwind utilities;

<!-- other style -->
```

Looking at the above structure, the styles in sds are used repeatedly in accounts, so @tailwind [base|components|utilites] is duplicated and overlaid with the same style.

So what I want is to know how to remove overlapping styles in these cases

--- 

Q2.
Additionally, the way I solved it was duplicated by removing the @tailwind [base|components] directive to accounts.

However, if an instruction such as '@layerbase' is used in accounts, an error such as '@layerbase' is used but no matching '@tailwindbase' directive is present.' It seems that the '@tailwindbase' instruction must be used to use the '@layerbase' instruction.

How can we solve this problem?

**Comment by florence0239** [comment]
Hello,
Q1: Removing Overlapping Styles
To avoid duplicate styles in your monorepo:

Centralize Tailwind Config: Create a shared tailwind.config.js in the root:
`module.exports = {
  content: ['./apps/**/*.{js,ts,jsx,tsx}', './packages/**/*.{js,ts,jsx,tsx}'],
  theme: { extend: {} },
  plugins: [],
};`
Shared CSS File: Create shared-styles.css:
`@tailwind base;
@tailwind components;
@tailwind utilities;`
Import Shared Styles:
`/* apps/accounts/styles/global.css */
@import '../../../shared-styles.css';
@import '../components/sds/input/InputField/input.style.css';`
`/* packages/sds/styles/global.css */
@import '../../../shared-styles.css';`
Q2: Using @layer Directives
Ensure @tailwind base is included in your shared styles to use @layer directives:

Define Layers in Shared Styles
`@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base { /* Custom base styles */ }
@layer components { /* Custom component styles */ }
@layer utilities { /* Custom utility styles */ }`
Use Layers in Accounts:
`@import '../../../shared-styles.css';
@import '../components/sds/input/InputField/input.style.css';

@layer base { /* Custom base styles for accounts */ }
@layer components { /* Custom component styles for accounts */ }
@layer utilities { /* Custom utility styles for accounts */ }`
This should help streamline your styles and avoid duplication.


Best regards,
florence0239
[Official Site](https://www-choiceadvantage.com)

**Comment by KIMSEUNGGYU** [comment]
[self solve]

I found a similar problem in stackoverflow and solved it [stackoverflow-post](https://stackoverflow.com/questions/72730384/layer-utilities-is-used-but-no-matching-tailwind-utilities-directive-is-presen)

I solved it by adding 'postcss-import' plug-in to postcss.config.ts
```ts
module.exports = {
  plugins: {
    'postcss-import': {}, // add
    tailwindcss: {},
    autoprefixer: {},
  },
};

```
